pkg/webhooks/podimagespec/podimagespec.go

```diff
@@ -0,0 +1,274 @@
+package podimagespec
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/openshift/managed-cluster-validating-webhooks/pkg/k8sutil"
+	"github.com/openshift/managed-cluster-validating-webhooks/pkg/webhooks/utils"
+
+	configv1 "github.com/openshift/api/operator/v1"
+	admissionregv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	admissionctl "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	WebhookName string = "podimagespec-mutation"
+	docString   string = `OpenShift debugging tools on Managed OpenShift clusters must be available even if internal image registry is removed.`
+)
+
+var (
+	timeout int32 = 2
+	scope         = admissionregv1.NamespacedScope
+	rules         = []admissionregv1.RuleWithOperations{
+		{
+			Operations: []admissionregv1.OperationType{
+				admissionregv1.Create,
+			},
+			Rule: admissionregv1.Rule{
+				APIGroups:   []string{""},
+				APIVersions: []string{"v1"},
+				Resources:   []string{"pods"},
+				Scope:       &scope,
+			},
+		},
+	}
+	log        = logf.Log.WithName(WebhookName)
+	imageRegex = regexp.MustCompile(`(image-registry\.openshift-image-registry\.svc:5000\/)(?P<namespace>\S*)(/)(?P<image>\S*)(:)(?P<tag>\S*)`)
+	client     = k8sutil.MustHaveContainerClient()
+)
+
+// PodImageSpecWebhook mutates an image spec in a pod
+type PodImageSpecWebhook struct {
+	s      runtime.Scheme
+	Client client.Client
+}
+
+// NewWebhook creates the new webhook
+func NewWebhook() *PodImageSpecWebhook {
+	scheme := runtime.NewScheme()
+	return &PodImageSpecWebhook{
+		s:      scheme,
+		Client: client.Client,
+	}
+}
+
+// CheckImageRegistryStatus checks the status of the image registry service
+func (s *PodImageSpecWebhook) CheckImageRegistryStatus(ctx context.Context) (bool, error) {
+	config := &configv1.Config{}
+	err := s.Client.Get(ctx, client.ObjectKey{
+		Namespace: "openshift-image-registry",
+		Name:      "cluster",
+	}, config)
+	if err != nil {
+		return false, err
+	}
+
+	if config.Spec.ManagementState == "Managed" {
+		return true, nil
+	} else if config.Spec.ManagementState == "Removed" {
```

Comment:
Use the proper type object https://github.com/openshift/api/blob/master/imageregistry/v1/types.go#L25-L31

Reply:
operatorv1.Management.Removed ,instead of string "Removed"